Introduce RFC000 - Process description (#28)
comit-network/RFCs: RFC-000-Process-description.md

# Process description

- RFC-Number: 000
- Status: Draft
- Discussion-issue: -
- Created on: 25 Jan. 2019

> <!-- TOC GOES here -->

Reviewer: don't forget the TOC :)

## Description

> This document describes the RFC (Request-for-comments) process that is used to document and specify the COMIT procotol.

Reviewer: ❌ 
```suggestion
This document describes the RFC (Request For Comments) process that is used to document and specify the COMIT protocol.
```

> The reason for using an RFC process for this is that we would like the documentation itself and the process around it to be open-source and accessible to anyone.

Reviewer: 🙃 
```suggestion
We are using an RFC process because we want the documentation and the process around it to be open-source and accessible to anyone.
```

## Lifecycle

### Everything starts as `Draft`

Anybody is welcome to submit an RFC to extend or change the COMIT protocol by opening a pull-request to this repository.

> All accepted changes will have status `Draft`.

Reviewer: ❓ what do you mean? Do you mean all merged Pull Requests?

Author: Yes. A merged PR = change to the protocol.

> RFCs in status `Draft` MUST have a discussion issue that can be used to discuss the current state of the RFC.

Reviewer: 🙃 
```suggestion
RFCs in `Draft` status MUST have a discussion issue that can be used to discuss the current content of the RFC.
```

> This will be the place where anybody can voice opinions or point out possible flaws or inconsistencies.

Reviewer: 🙃 better phrasing?

Author: suggestions?

As instructed by the [issue template](./.github/ISSUE_TEMPLATE/discussion_issue.md), please add a checklist for unresolved problems or open questions at the end of the discussion issue.

> As long as an RFC is in `Draft`, new PRs can change it accordingly to reflect the newest findings of the discussions.

Reviewer: 🙃 
```suggestion
As long as an RFC is in `Draft`, new PRs can change it accordingly to reflect the newest findings of from the discussion issue.
```

Author: I disagree, it is about the findings from the discussions which just happen to be in an issue for now but could also happen outside of it.

### On to stabilization

> As soon as all the contributors are happy with the RFC, a stabilization PR can be opened.

Reviewer: ❓ define happy? do we want to set a time period? (ie, no more comment has been made for one month)

Author: Good point. I'll make that an open question.

Author: @comit-network/rust-devs opinions?

Reviewer 2: Yeah, that's pretty vague. I like the period without comment. I guess there's never gonna be a reason to rush an RFC out the door anyway.
